llvm/include/llvm/MC/CAS/MCCASObjectV1.h

```diff
@@ -6,6 +6,99 @@
 //
 //===----------------------------------------------------------------------===//
 
+/// \verbatim
+/// CAS Schema:
+///
+/// Debug Line Section
+///
+/// DebugLineSectionRef: The DebugLineSectionRef represents the debug_line
+/// section, it contains the relocations for that section as well as any addends
+/// and their offsets and sizes, that were zero-ed out for deduplication
+/// purposes.
+///
+/// DebugLineRef: The DebugLineRef represents one function's contribution to the
+/// debug_line section, or one line table. Each function will have it's own line
+/// table entry in the debug_line section.
+///
+/// ┌─────────────────┐
+/// │  DebugLineRef   │
+/// │  [Line Table 1] │
+/// │                 │         ┌──────────────────────┐
+/// └──────────▲──────┘         │  DebugLineSectionRef │
+///       .    ├────────────────┤                      │
+///       .    │                │     [Relocations]    │
+///       .    │                │     [Addends]        │
+/// ┌──────────▼──────┐         └──────────────────────┘
+/// │  DebugLineRef   │
+/// │  [Line Table N] │
+/// │                 │
+/// └─────────────────┘
+///
+/// Debug String Section
+///
+/// DebugStringSectionRef: The DebugStringSectionRef represents the debug_str
+/// section.
+///
+/// DebugStrRef: A DebugStrRef represents one string in the debug_str section,
+/// each string gets its own cas object.
+///                         ┌────────────────┐
+///            ┌────────────┤  DebugString   ├───────────┐
+///            │            │  SectionRef    │           │
+///            │            └────────────────┘           │
+///     ┌──────▼───────┐                         ┌───────▼──────┐
+///     | DebugStrRef  │                         │ DebugStrRef  |
+///     │ [String 1]   │          ...            │ [String N]   │
+///     └──────────────┘                         └──────────────┘
+///
+///
+/// Debug Info Section
+///
+/// DebugInfoSectionRef: The DebugInfoSectionRef represents the debug_line
+/// section, it contains the relocations for that section as well as any addends
+/// and their offsets and sizes, that were zero-ed out for deduplication
+/// purposes.
+///
+/// DistinctDataRef: The DistinctDataRef is one CAS block per translation unit
+/// that contains all the information from all compile units that does not
+/// deduplicate.
+///
+/// AbbrevOffsetRef: The AbbrevOffsetRef is one CAS block per translation unit
+/// that contains all the abbreviation offsets of every compile unit in that
+/// translation unit.
+///
+/// DebugInfoCURef: A DebugInfoCURef represents one function's contribution to
+/// the debug_info section, i.e. one compile unit. We have changed the meaning
```

Comment:
We have -> When compiling with -fcas-friendly each function is stored in its own compile unit.